src/include/function/aggregate/collect.h

```diff
@@ -0,0 +1,147 @@
+        if (input->hasNoNullsGuarantee()) {
+            for (auto i = 0u; i < input->state->selVector->selectedSize; ++i) {
+                auto pos = input->state->selVector->selectedPositions[i];
+                updateSingleValue(state, input, pos, multiplicity, memoryManager);
+            }
+        } else {
+            for (auto i = 0u; i < input->state->selVector->selectedSize; ++i) {
+                auto pos = input->state->selVector->selectedPositions[i];
+                if (!input->isNull(pos)) {
+                    updateSingleValue(state, input, pos, multiplicity, memoryManager);
+                }
+            }
+        }
+    }
+
+    static inline void updatePos(uint8_t* state_, common::ValueVector* input, uint64_t multiplicity,
+        uint32_t pos, storage::MemoryManager* memoryManager) {
+        auto state = reinterpret_cast<CollectState*>(state_);
+        updateSingleValue(state, input, pos, multiplicity, memoryManager);
+    }
+
+    static void initCollectStateIfNecessary(
+        CollectState* state, storage::MemoryManager* memoryManager, common::DataType& dataType) {
+        if (state->factorizedTable == nullptr) {
+            auto tableSchema = std::make_unique<processor::FactorizedTableSchema>();
+            tableSchema->appendColumn(
+                std::make_unique<processor::ColumnSchema>(false /* isUnflat */,
+                    0 /* dataChunkPos */, common::Types::getDataTypeSize(dataType)));
+            state->factorizedTable =
+                std::make_unique<processor::FactorizedTable>(memoryManager, std::move(tableSchema));
+        }
+    }
+
+    static void updateSingleValue(CollectState* state, common::ValueVector* input, uint32_t pos,
+        uint64_t multiplicity, storage::MemoryManager* memoryManager) {
+        initCollectStateIfNecessary(state, memoryManager, input->dataType);
+        for (auto i = 0u; i < multiplicity; ++i) {
+            auto tuple = state->factorizedTable->appendEmptyTuple();
+            state->isNull = false;
+            common::ValueVectorUtils::copyNonNullDataWithSameTypeOutFromPos(
+                *input, pos, tuple, *state->factorizedTable->getInMemOverflowBuffer());
+        }
+    }
+
+    static void combine(uint8_t* state_, uint8_t* otherState_) {
+        auto otherState = reinterpret_cast<CollectState*>(otherState_);
+        if (otherState->isNull) {
+            return;
+        }
+        auto state = reinterpret_cast<CollectState*>(state_);
+        if (state->isNull) {
+            state->factorizedTable = std::move(otherState->factorizedTable);
+            state->isNull = false;
+        } else {
+            state->factorizedTable->merge(*otherState->factorizedTable);
+        }
+    }
+
+    static void finalize(uint8_t* state_) {}
+
+    static void collectBindFunc(const std::vector<common::DataType>& argumentTypes,
+        FunctionDefinition* definition, common::DataType& actualReturnType) {
+        if (argumentTypes.size() != 1) {
```

Comment:
Instead of checking with if statement, I think it's better to just assert. Because binder should guarantee you get a match with exactly 1 parameter.

```
assert(argumentTypes.size() == 1)
auto inputType = argumentTypes[0];
```

Reply:
changed to assertion